### PR TITLE
Fix for issue #19020

### PR DESCRIPTION
**Pull request by adi928**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

Fixes the issue where facebook redirects to a warning page, causing youtube-dl to fail, instead of redirecting to the external link. Fix for issue #19020.

### CODE REVIEW

File: youtube_dl/extractor/facebook.py [hunk 1]
```diff
@@ -39,15 +39,16 @@ class FacebookIE(InfoExtractor):
                                 photo\.php|
                                 video\.php|
                                 video/embed|
-                                story\.php
-                            )\?(?:.*?)(?:v|video_id|story_fbid)=|
+                                story\.php|
+                                flx/warn/
+                            )\?(?:.*?)(?:u|v|video_id|story_fbid)=|
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: No.

File: youtube_dl/extractor/facebook.py [hunk 2]
```diff
@@ -39,15 +39,16 @@ class FacebookIE(InfoExtractor):
...
                             [^/]+/videos/(?:[^/]+/)?|
                             [^/]+/posts/|
                             groups/[^/]+/permalink/
                         )|
                     facebook:
                 )
-                (?P<id>[0-9]+)
+                (?P<id>[0-9]+|https://.*)
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: No.

File: youtube_dl/extractor/facebook.py [hunk 3]
```diff
@@ -455,6 +456,13 @@ def _real_extract(self, url):
         if info_dict:
             return info_dict
 
+        # In case the url is a redirect and contains an external link.
+        if video_id.startswith("https://"):
+            return {
+                        '_type': 'url',
+                        'url': video_id,
+                    }
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: Must be separate extractor.

Comment (adi928) [reply youtube_dl/extractor/facebook.py]: So a separate new extractor with a regex for the Facebook 'warning' URL? Any suggestions for the name for this new extractor? I was going to go with facebookWarning extractor.

Also, in the new extractor, won't we be ultimately passing the redirected URL to the generic.py, as the code is doing here?

Comment (dstftw) [reply youtube_dl/extractor/facebook.py]: Name essentially does not matter. `FacebookExternalLinkIE` is a reasonable name. "The code here" does not pass to the generic extractor in the first place.

Comment (adi928) [reply youtube_dl/extractor/facebook.py]: I'm sorry, I meant 'common.py'. Then would that be fine to pass the link to common.py for download?

Comment (dstftw) [reply youtube_dl/extractor/facebook.py]: It's already fine. More idiomatic way is to use `url_result`.

Comment (adi928) [reply youtube_dl/extractor/facebook.py]: Created another PR (#19116) with the correct branch as per the developer instructions.

**Pull request closed without merging**